pkg/base/helm.go

```diff
@@ -235,13 +235,26 @@ func removeCommonPrefix(baseFiles []BaseFile) []BaseFile {
 
 func helmChartBaseAppendAdditionalFiles(base Base, u *upstreamtypes.Upstream) Base {
 	for _, upstreamFile := range u.Files {
-		if upstreamFile.Path == path.Join(base.Path, "Chart.yaml") {
+		// need to check if base path is an empty string here to catch just the top level additional files
+		// otherwise, we need to check if the upstream path contains the base path to correctly add files for sub-charts
+		if base.Path == "" && upstreamFile.Path == path.Join(base.Path, "Chart.yaml") {
```

Comment:
nit pick: i would do it the "go" way, something like this at the beginning of the loop
```
if base.Path != "" {
  continue
}
```